src/LambdaLauncher/Plugins/Sway.hs

```diff
@@ -31,17 +31,17 @@ instance ToJSON Node
 findWindows :: Node -> [Window]
 findWindows ((Node (Just name) app_id id (Just pid) _)) =
   [Window name app_id id pid]
-findWindows (node) = concat (mconcat <$> map findWindows <$> nodes node)
+findWindows node = concat (mconcat . map findWindows <$> nodes node)
 
 windowToResults :: Window -> LambdaLauncher.Types.Result
 windowToResults (Window name app_id id pid) =
   Action name 2 $ callProcess "swaymsg" ["[con_id=" ++ show id ++ "] focus"]
 
 sway :: Plugin
 sway s = do
-  tree <- encodeUtf8 <$> pack <$> readProcess "swaymsg" ["-t", "get_tree"] ""
-  return $
-    concat $
-    map windowToResults <$> filter (\(Window name _ _ _) -> isInfixOf s name) <$>
```

Comment:
It is better to make labels for Window type and use here `Window{..}`

Reply:
I don`t know what fields mean to make it myself.

Reply:
Good point. It probably should be 
```
data Window =
  Window 
         { name :: String
         , app_id :: Maybe String
         , id :: Integer
         , pid :: Integer }
```

Reply:
Ok, i will add.